README.md

```diff
@@ -1635,6 +1659,267 @@ map.addEntryListener(mapEventListener).then(function () {
+
+- `between`: Checks if the result of an expression is between two values (this is inclusive).
+
+- `inPredicate`: Checks if the result of an expression is an element of a certain list.
+
+- `not`: Checks if the result of an expression is false.
+
+- `regex`: Checks if the result of an expression matches some regular expression.
+
+Hazelcast offers the following ways for distributed query purposes:
+
+- Combining Predicates with AND, OR, NOT
+
+- Distributed SQL Query
+
+#### 7.7.1.1. Employee Map Query Example
+
+Assume that you have an `employee` map containing values of `Employee` objects, as coded below. 
+
+```javascript
+function Employee(name, age, active, salary) {
+    this.name = name;
+    this.age = age;
+    this.active = active;
+    this.salary = salary;
+}
+
+Employee.prototype.getClassId = function () {
+    return 1;
+}
+
+Employee.prototype.getFactoryId = function () {
+    return 1;
+}
+
+Employee.prototype.readData = function (objectDataInput) {
+    this.name = objectDataInput.readUTF();
+    this.age = objectDataInput.readInt();
+    this.active = objectDataInput.readBoolean();
+    this.salary = objectDataInput.readDouble();
+};
+
+Employee.prototype.writeData = function (objectDataOutput) {
+    objectDataOutput.writeUTF(this.name);
+    objectDataOutput.writeInt(this.age);
+    objectDataOutput.writeBoolean(this.active);
+    objectDataOutput.writeDouble(this.salary);
+}
+```
+
+Note that `Employee` is an `IdentifiedDataSerializable` object. If you just want to save the `Employee` objects as byte arrays on the map, you don't need to implement its equivalent on the server-side. However, if you want to query on the `employee` map, server needs the `Employee` objects rather than byte array formats. Therefore, you need to implement its Java equivalent and its data serializable factory on server side for server to reconstitute the objects from binary formats. After implementing the Java class and its factory, you need to add the factory to the data serializable factories or the portable factories by giving a factory id. Here is the example XML configuration of the server.
+
+```xml
+<hazelcast>
+    ...
+    <serialization>
+        <data-serializable-factories>
+            <data-serializable-factory factory-id="1">
+                mypackage.MyIdentifiedFactory
+            </data-serializable-factory>
+        </data-serializable-factories>
+    </serialization>
+    ...
+</hazelcast>
+```
+
+Note that before starting the server, you need to compile `Employee` and `MyIdentifiedFactory` classes with server's `CLASSPATH` and add them to `user-lib` folder in the extracted `hazelcast-<version>.zip`. See [Adding User Library to CLASSPATH section](#adding-user-library-to-classpath).
+
+> **NOTE: You can also make this object `Portable` and implement its Java equivalent and its portable factory on server side. Note that querying with `Portable` object is faster than `IdentifiedDataSerializable`.**
```

Comment:
... on the server side. Note that querying with `Portable` object is faster as compared to `IdentifiedDataSerializable`.